kratos/python/add_other_utilities_to_python.cpp

```diff
@@ -53,6 +53,23 @@
 namespace Kratos {
 namespace Python {
 
+/**
+ * @brief A thin wrapper for GetSortedListOfFileNameData. The reason for having the wrapper is to replace the original lambda implementation as it causes gcc 4.8 to generate bad code on Centos7 which leads to memory corruption.
+ */   
+pybind11::list GetSortedListOfFileNameDataHelper(
+    std::vector<FileNameDataCollector::FileNameData>& rFileNameDataList,
+    const std::vector<std::string> & rSortingFlagsOrder
+    )
+{
+    FileNameDataCollector::SortListOfFileNameData(rFileNameDataList, rSortingFlagsOrder);
+    pybind11::list result;
+    for (unsigned int j = 0; j < rFileNameDataList.size(); j++)
```

Comment:
```suggestion
    for (unsigned int j = 0; j < rFileNameDataList.size(); ++j)
```

Reply:
I always ask the same but Is there any specific reason to change to pre here? This is not a complex type and the compiler can handle the temp variable. Also, note that in parallel regions is always better to use post because it does not force the compiler to introduce a data dependency.

Reply:
I don't think there is a performance gain by having pre or post (As you mentioned it will be clearly identified by the compiler), But what I was suggesting is to have only to show that, eventhough you may say j++, compiler will do ++j. Or am I wrong?

> Also, note that in parallel regions is always better to use post because it does not force the compiler to introduce a data dependency.

I completely agree :) (You are the expert :) )

Reply:
> I don't think there is a performance gain by having pre or post 

Ups, sorry I just assumed that myself.

> But what I was suggesting is to have only to show that, even though you may say j++, compiler will do ++j. Or am I wrong?

That I cannot tell. I would expect that for a simple loop like that the compiler just ignores the variable and operates directly over `&rFileNameDataList.data()` but ofc this will depend on the optimization level, the structure you are iterating, the compiler and so on and so forth... That's why I always make the same comment when I see the `++i` suggestion. Unless you are very picky about the side effects the operator gives no clue about what the compiler may be doing. For me is more a matter of taste in this scenario and provided that most loops in kratos are post incremented